packages/common/src/dotlottie-player.ts

```diff
@@ -137,6 +159,17 @@ export class DotLottiePlayer {
     };
   }
 
+  protected _extractExtraOptions(config: DotLottieConfig<RendererType>): ExtraOptions {
+    return Object.entries(config).reduce((newConfig, [key, value]) => {
+      if (!Object.hasOwn(EXTRA_OPTIONS, key)) return newConfig;
+
+      return {
+        ...newConfig,
+        [key]: value,
+      };
+    }, EXTRA_OPTIONS);
```

Comment:
it needs extra cognitive load to process the purpose of using `reduce` in this function, i wonder if it's possible to use normal `forEach` or `for...of` to improve the readability of the code. what do you think ?

```suggestion
   const extraOptions: Partial<ExtraOptions> = {};
  
  for (const [key, value] of Object.entries(config)) {
    ....
  }

  return extraOptions
```